File: SwiftProject/SwiftProject/K02.swift
//
//  K02.swift
//  SwiftProject
//
//  Created by 周哲輝 on 2019/05/05.
//  Copyright © 2019 周哲輝. All rights reserved.
//

import Foundation

public class K02 : IExecutable {
    public override func Question1() {
        let seisuu = 3
        let jissuu = 2.6
        let moji = "A"
        
        print("整数seisuuの値は\(seisuu)")
        print("実数jissuuの値は\(jissuu)")
        print("文字mojiの値は\(moji)")
    }
    
    public override func Question2() {
        print("一つ目の整数は？")
        let firstNumber = Int(readLine()!)!
        print("二つ目の整数は？")
        let secondNumber = Int(readLine()!)!
        
        print("\(firstNumber)÷\(secondNumber)=\(firstNumber / secondNumber)...\(firstNumber % secondNumber)")
    }
    
    public override func Question3() {
        print("一つ目の商品の値段は？")
        let priceA = Int(readLine()!)!
        print("個数は？")
        let amountA = Int(readLine()!)!
        print("二つ目の商品の値段は？")
        let priceB = Int(readLine()!)!
        print("個数は？")
        let amountB = Int(readLine()!)!
        
        let total = Float((priceA * amountA + priceB * amountB)) * 1.08
        print("お支払いは税込み￥\(total)です。")
    }
    
    public override func Question4() {
        
        var golem = Golem()
        
        print("ゴーレム　（HP：\(golem.hp) 防御力：\(golem.defense)）")
        print("HP：\(golem.hp)")
        print("今回の攻撃の値を入力してください＞")
        var damage = Int(readLine()!)!
        
        damage = damage - golem.defense > 0 ? damage - golem.defense : 0
        print("ダメージは\(damage)です。")
        golem.hp -= damage
        print("残りのHPは\(golem.hp)です。")
    }
    
    
}

File: Scala/src/main/scala/PrimaryClasses/K02.scala
package PrimaryClasses
import Golem.Golem
import IExecutable.IExecutable

class K02 extends IExecutable {
  override def Question1(): Unit = {
    val seisuu = 3
    val jissuu: Float = 2.6f
    val moji = 'A'

    printf("変数seisuuの値は%d\n", seisuu)
    printf("変数jissuuの値は%f\n", jissuu)
    printf("変数mojiの値は%c\n", moji)
  }

  override def Question2(): Unit = {
    print("一つ目の整数は？")
    val firstNumber = scala.io.StdIn.readInt
    print("二つ目の整数は？")
    val secondNumber = scala.io.StdIn.readInt

    printf("%d÷%d=%d...%d", firstNumber, secondNumber, firstNumber / secondNumber, firstNumber % secondNumber)
  }

  override def Question3(): Unit = {
    print("一つ目の商品の値段は？")
    val priceA = scala.io.StdIn.readInt
    print("個数は？")
    val amountA = scala.io.StdIn.readInt
    print("二つ目の商品の値段は？")
    val priceB = scala.io.StdIn.readInt
    print("個数は？")
    val amountB = scala.io.StdIn.readInt

    val total: Float = (priceA * amountA + priceB * amountB) * 1.08f
    printf("お支払いは税込み￥%fです", total)
  }

  override def Question4(): Unit = {
    var golem = new Golem

    printf("ゴーレム　（HP:%d　防御力：%d）\n", golem.hp, golem.defense)
    printf("HP：%d\n", golem.hp)
    print("今回の攻撃の値を入力してください＞")
    var damage = scala.io.StdIn.readInt()
    damage = if (damage - golem.defense > 0) damage - golem.defense else 0
    printf("ダメージは%dです。\n", damage)
    golem.hp -= damage
    printf("残りのHPは%dです。\n", golem.hp)
  }
}
